dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx

```diff
@@ -222,20 +222,10 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
     );
   };
 
-  const isLineOrScatter = (vs: string) => (vs === visChartTypes.Line || vs === visChartTypes.Scatter);
-  const isVerticalorHorizontalBar = (vs: string) => (vs === visChartTypes.Bar || vs === visChartTypes.HorizontalBar);
-
   const memorizedVisualizationTypes = useMemo(() => {
     let visDefinition = {}
     return ENABLED_VIS_TYPES.map((vs: string) => {
-      if (isLineOrScatter(vs)) {
-        visDefinition = vs === visChartTypes.Line ? getVisType(vs, { type: visChartTypes.Line }) : getVisType(vs, { type: visChartTypes.Scatter });
-      } else if (isVerticalorHorizontalBar(vs)) {
-        visDefinition = vs === visChartTypes.Bar ? getVisType(vs, { type: visChartTypes.Bar }) : getVisType(vs, { type: visChartTypes.HorizontalBar });
-      } else {
-        visDefinition = getVisType(vs);
-      }
-
+      visDefinition = getVisTypeData(vs);
```

Comment:
This `visDefinition` is going to be updated for each of the entry in the `ENABLED_VIS_TYPES` array. So you are only going to get the last one. In addition, your map function is not returning anything. Take a look. If you need the last element only, there is no need for a loop.

Reply:
Optimized above block of code as per your suggestions.